src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -194,64 +213,80 @@ export function removeItem(
   });
 }
 
+function setEquipForItemInLoadout(
+  defs: D1ManifestDefinitions | D2ManifestDefinitions,
+  item: DimItem,
+  loadout: Loadout,
+  isEquipped: boolean
+) {
+  // Subclasses and some others are always equipped
+  if (singularBucketHashes.includes(item.bucket.hash)) {
+    return;
+  }
+
+  // TODO: it might be nice if we just assigned a unique ID to every loadout item just for in-memory ops like deleting
+  // We can't just look it up by identity since Immer wraps objects in a proxy and getItemsFromLoadoutItems
+  // changes the socketOverrides, so simply search by unmodified ID and hash.
+  const loadoutItemIndex = findSameLoadoutItemIndex(
+    defs,
+    loadout.items,
+    convertToLoadoutItem(item, false, 1)
+  );
+
+  if (loadoutItemIndex === -1) {
+    return;
+  }
+  const loadoutItem = loadout.items[loadoutItemIndex];
+
+  if (item.equipment) {
+    if (!isEquipped) {
+      // It's equipped, mark it unequipped
+      loadoutItem.equip = false;
+    } else {
+      // It's unequipped - mark all the other items in the same bucket, and conflicting exotics, as unequipped unequipped, then mark this equipped
+      for (const li of loadout.items) {
+        const itemDef = defs.InventoryItem.get(li.hash);
+        const bucketHash =
+          itemDef &&
+          ('bucketTypeHash' in itemDef
+            ? itemDef.bucketTypeHash
+            : itemDef.inventory?.bucketTypeHash);
+
+        const equippingLabel =
+          itemDef && 'tierType' in itemDef
+            ? itemDef.tierType === TierType.Exotic
+              ? itemDef.itemType.toString()
+              : undefined
+            : itemDef.equippingBlock?.uniqueLabel;
+
+        // Others in this slot
+        if (
+          bucketHash === item.bucket.hash ||
+          // Other exotics
+          (item.equippingLabel && equippingLabel === item.equippingLabel)
+        ) {
+          li.equip = false;
+        }
+      }
+      loadoutItem.equip = true;
+    }
+  }
+}
+
```

Comment:
Extracted out of the `equipItem` action for this so it can be used in a few places.